File: java/dirty/src/main/java/fr/cleancode/Block.java
package fr.cleancode;

import org.apache.commons.codec.digest.DigestUtils;

class Block {
    public String hash;
    public String data;
    public String parentHash;

    //This constructor create a block with a parentHash and some data
    Block(String parentHash, String data) {
        this.data = data;
        this.hash = parentHash == null ? DigestUtils.sha256Hex(data) : DigestUtils.sha256Hex(parentHash + data);
        this.parentHash = parentHash == null ? hash : parentHash;
    }
}

File: js/dirty/src/Block.js
const crypto = require("crypto");

class Block {
    //This constructor create a block with a parentHash and some data
    constructor(parentHash, data) {
        this.data = data;
        this.hash = Block.makePuzzle(
            this.parentHash,
            this.data
        );
        this.parentHash = parentHash === null ? this.hash : parentHash;
    }

    static makePuzzle(parentHash, data) {
        return crypto
            .createHash("sha256")
            .update(parentHash || "")
            .update(JSON.stringify(data) || "")
            .digest("hex");
    }
    
}

module.exports = Block;
